### PR TITLE
Migrates fix for messageformat patching

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Fixes issue where translated elements on the page could end up in a different language when this language received an update. 
Refactors some translations-centric classes and adds unit tests. 
Cleans up app.component and migrates remaining pieces, except Telemetry, which will be done in the performance branch. 

medic/angular10-migration#122
medic/angular10-migration#119
medic/angular10-migration#118

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
@latin-panda could you, please, have a look? Thanks!

File: webapp/src/ts/providers/translation-loader.provider.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
 import { Injectable } from '@angular/core';
 import { TranslateLoader } from '@ngx-translate/core';
 import { from } from 'rxjs';
 
 import { DbService } from '../services/db.service';
+import * as translationUtils from '@medic/translation-utils';
+import { TranslationDocsMatcherProvider } from '@mm-providers/translation-docs-matcher.provider';
 
 @Injectable()
 export class TranslationLoaderProvider implements TranslateLoader {
```
Comment (latin-panda) [inline webapp/src/ts/providers/translation-loader.provider.ts:10]: Can you please add a dedicated unit test file for this provider with error and success cases?

**Review by latin-panda — COMMENTED** [review]
Awesome 👏  
I added a couple of questions in line

File: webapp/custom-webpack.config.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ module.exports = {
       // https://github.com/enketo/enketo-core/issues/454
       '../../js/dropdown.jquery': 'node_modules/bootstrap/js/dropdown',
       'bikram-sambat': 'node_modules/bikram-sambat',
+      'messageformat': 'node_modules/messageformat/index',
```
Comment (latin-panda) [inline webapp/custom-webpack.config.js:17]: Do we need to do something else to integrate this one?

Comment (dianabarsan) [reply webapp/custom-webpack.config.js:17]: This should be enough.

File: webapp/src/ts/providers/translation-docs-matcher.provider.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class TranslationDocsMatcherProvider {
+  private static DOC_ID_PREFIX = 'messages-';
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply webapp/src/ts/providers/translation-docs-matcher.provider.ts:5]: yes

File: webapp/src/ts/providers/translation-loader.provider.ts (hunk 1 above)
Comment (dianabarsan) [reply webapp/src/ts/providers/translation-loader.provider.ts:10]: Sure thing :)

File: webapp/src/ts/providers/translation-docs-matcher.provider.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  private static translationsDocIdMatcher = new RegExp(`^${TranslationDocsMatcherProvider.DOC_ID_PREFIX}(.+)$`);
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply webapp/src/ts/providers/translation-docs-matcher.provider.ts:6]: > or make the RegExp inside the functions that need it
that would mean that the regexp would need to be compiled for every execution, which is not entirely desirable :)

File: webapp/src/ts/providers/translation-loader.provider.ts (hunk 1 above)
Comment (dianabarsan) [reply webapp/src/ts/providers/translation-loader.provider.ts:10]: There already is a unit test: https://github.com/medic/cht-core/blob/4750-angular-10-take-2/webapp/tests/karma/ts/providers/translation-loader.provider.spec.ts
Does it suffice?

**Comment by dianabarsan** [comment]
hey @latin-panda I think I addressed all your questions. Please have another look.

**Comment by latin-panda** [comment]
Awesome looks good! Thank you

**Pull request merged**